Lib/multiprocessing/pool.py

```diff
@@ -429,6 +449,16 @@ def _map_async(self, func, iterable, mapper, chunksize=None, callback=None,
         )
         return result
 
+    def _wait_for_updates(self, timeout=None):
+        task_queue_sentinels = [self._outqueue._reader.fileno(),
+                                self._outqueue._writer.fileno()]
+        self_notifier_sentinels = [self._change_notifier._reader.fileno()]
+        worker_sentinels = [worker.sentinel for worker in self._pool]
+        sentinels = [*task_queue_sentinels,
+                     *worker_sentinels,
+                     *self_notifier_sentinels]
+        wait(sentinels, timeout=timeout)
```

Comment:
Don't you have to consome the _change_notifier queue somewhere, to avoid to get a full queue or just to avoid wasting memory? Call non-blocking get() until it raises an empty error? I suggest to do that after the wait().